mne/preprocessing/ica.py

```diff
@@ -1093,9 +1086,6 @@ def find_bads_eog(self, inst, ch_name=None, threshold=3.0,
         # get targets befor equalizing
         targets = [self._check_target(k, inst, start, stop) for k in eog_chs]
 
-        if inst.ch_names != self.ch_names:
-            inst = inst.copy().pick_channels(self.ch_names)
```

Comment:
I'm not sure out tests fully cover the consequences. The reason we need the copy here is that we need to pick in certain situations and then we don't want to modify the data that are passed here. If you want to change this you need to make sure you test well passing instances with different channels than ICA is fitted on and also deal with the ValueErrors we put here.